app/routes/datasets/versions.py

```diff
@@ -106,8 +108,11 @@ async def add_new_version(
 
     input_data["creation_options"] = creation_options
 
-    # Everything else happens in the background task asynchronously
-    background_tasks.add_task(create_default_asset, dataset, version, input_data, None)
+    if creation_options.get("delete_version", None):
```

Comment:
Is this to make an asset record for delete operations? Think I overlooked this, thanks!